cmd/agent/app/flags.go

```diff
@@ -24,54 +24,71 @@ import (
 	"flag"
 	"fmt"
 	"strings"
+	"github.com/spf13/viper"
 )
 
+type stringSliceFlag struct {
+	slice *[]string
+}
+
 // Bind binds the agent builder to command line options
-func (b *Builder) Bind(flags *flag.FlagSet) {
+func (b *Builder) GenerateFlags() *flag.FlagSet {
+	flagSet := &flag.FlagSet{}
+
 	for i := range b.Processors {
 		p := &b.Processors[i]
 		name := "processor." + string(p.Model) + "-" + string(p.Protocol) + "."
-		flags.IntVar(
-			&p.Workers,
+		flagSet.Int(
 			name+"workers",
 			p.Workers,
```

Comment:
hm... the reason we did it with flags & pointers was because in our internal `main()`s we populate Builder fields by reading YAML config files, and the flags are allowed to override the same fields. It was a fairly clean model as the command line property names are declared once via `flag` directives, the YAML names are declared in the struct. There is no duplication of strings as seems to happen now when you call `viper.GetInt(name+"server-max-packet-size")`